working/0546-patterns/patterns-feature-specification.md

```diff
@@ -2540,12 +2639,54 @@ To bind invocation keys in a pattern `p` using parent invocation `i`:
 
     1.  Bind `i : ("length", [])` to the `length` getter invocation.
 
-    2.  For each element subpattern:
+    2.  For each element subpattern `s`:
+
+        1.  If `s` is a rest element:
```

Comment:
Don't we need to distinguish between rest elements with a pattern (which call sublist) and those that don't?  That is, `[..., a]` doesn't cache anything for the sublist, but `[...rest, a]` does, right?

Reply:
Yeah, we don't have to worry about caching in this section. The point of this part of the proposal is to statically define a key to identify each cached object—or perhaps more precisely, to define which patterns *share* cached values— and we can harmlessly define keys for `sublist()` even for rest patterns that don't actually end up calling `sublist()`. It just means that *if* the rest pattern were to call `sublist()`, we know which other `sublist()` calls elsewhere in the set of cases would share the same cached value.